src/proformwifitreadmill.cpp

```diff
@@ -168,11 +198,15 @@ void proformwifitreadmill::characteristicChanged(const QString &newValue) {
 
     if (!values[QStringLiteral("Current KPH")].isUndefined()) {
         double kph = values[QStringLiteral("Current KPH")].toString().toDouble();
-        Speed = kph;
+        Speed = averagespeed(kph);
```

Comment:
The feedback of the machine is varying alot, and the apps, zwift or kinomap only ask once every few seconds to get the current speed. so, they end up with too low or too high speed value. Thats why I make an average here.

Reply:
The machine is not reporting the speed that you fix, but it has sensors that measure the actual speed of the "tapis" (in french) (probably treadmill in English). That is what is reported.
I could get you logs.

Reply:
Tapis is fine (we are calling here it too in Italy :) )
Hah now I got this! It's a wonderful thing (I mean have the real time speed instead of the one fixed).
So the best way to achieve this is to put this average under a new setting, leaving the default behaviour without the average. 

Also the metric class has already an average method, maybe it's better to use that one, don't you think?

Reply:
I haven't dived into so many details. You are definitely better placed to decide where this average gets implemented.

Reply:
ok no problem, now that I understood the issue :)

Reply:
@pssara could you please pull this and build it? I added a new setting under nordictrack treadmill called "Averaging Speed" (default disabled). So enable it and it should work as you want. Let me know, I'm waiting your confirmation before merging this!